client/hotrod-client/src/test/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelTimeoutTest.java

```diff
@@ -0,0 +1,94 @@
+package org.infinispan.client.hotrod.impl.transport.netty;
+
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.net.SocketAddress;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiConsumer;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.Cache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.retry.AbstractRetryTest;
+import org.infinispan.client.hotrod.retry.SocketTimeoutFailureRetryTest;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.client.hotrod.test.InternalRemoteCacheManager;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.interceptors.impl.EntryWrappingInterceptor;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.CleanupAfterTest;
+import org.testng.annotations.Test;
+
+@CleanupAfterTest
+@Test(groups = "functional", testName = "netty.ChannelTimeoutTest")
+public class ChannelTimeoutTest extends AbstractRetryTest {
```

Comment:
TBH, I don't think you need another test. You can modify `org.infinispan.client.hotrod.retry.SocketTimeoutFailureRetryTest#testRetrySocketTimeout` to check that the connection is not closed.